From c53b9533a3ce5495d6661a1166abdd019f80a1d0 Mon Sep 17 00:00:00 2001
From: Kexy Biscuit <kexybiscuit@aosc.io>
Date: Sun, 1 Sep 2024 00:15:17 +0800
Subject: [PATCH 1/2] main: Go back to text mode when quitting (if appropriate)

Since commit 48881ba2ef3d25fd27fd150d4d5957d4df9868e0 plymouth
goes into GRAPHICS mode early on. Unfortunately, there are cases
where it neglects to go back to TEXT mode when quitting. That can
happen if boot finishes before the splash screen is created.

This commit fixes that.

Ref: https://gitlab.freedesktop.org/plymouth/plymouth/-/commit/d2ab367e12423646d3a6bb35d16570f8e3126234
---
 src/main.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/src/main.c b/src/main.c
index 33fe51e..5916ddc 100644
--- a/src/main.c
+++ b/src/main.c
@@ -1223,10 +1223,8 @@ hide_splash (state_t *state)
 
         cancel_pending_delayed_show (state);
 
-        if (state->boot_splash == NULL)
-                return;
-
-        ply_boot_splash_hide (state->boot_splash);
+        if (state->boot_splash != NULL)
+                ply_boot_splash_hide (state->boot_splash);
 
         if (state->local_console_terminal != NULL) {
                 ply_terminal_set_mode (state->local_console_terminal, PLY_TERMINAL_MODE_TEXT);
-- 
2.46.1

